### ISSUE TITLE
Test failure with PHP 7.2.0-dev

### ISSUE THREAD

**Issue by remicollet** [Kdyby/Events#108]
```
+ php72 /usr/bin/nette-tester --colors 0 -p php72 -C tests/KdybyTests -s
 _____ ___  ___ _____ ___  ___
|_   _/ __)( __/_   _/ __)| _ )
  |_| \___ /___) |_| \___ |_|_\  v2.0.x

PHP 7.2.0-dev (cli) | php72 | 8 threads

.........................................F.....F............

-- FAILED: Kdyby\Events\Extension. | KdybyTests/Events/Extension.phpt [method=testAutowire]
   Exited with error code 255 (expected 0)
   E_DEPRECATED: The each() function is deprecated. This message will be suppressed on further calls
   
   in Nette/Http/RequestFactory.php(114) 
   in Tester/Framework/TestCase.php(124) call_user_func_array()
   in [internal function]Tester\TestCase->Tester\{closure}()
   in Nette/Http/RequestFactory.php(114) each()
   in cache/Nette.Configurator/Container_315f70ef90.php(183) Nette\Http\RequestFactory->createHttpRequest()
   in Nette/DI/Container.php(179) Container_315f70ef90->createServiceHttp__request()
   in Nette/DI/Container.php(105) Nette\DI\Container->createService()
   in cache/Nette.Configurator/Container_315f70ef90.php(103) Nette\DI\Container->getService()
   in Nette/DI/Container.php(179) Container_315f70ef90->createServiceApplication__application()
   in Nette/DI/Container.php(105) Nette\DI\Container->createService()
   in Nette/DI/Container.php(103) Nette\DI\Container->getService()
   in KdybyTests/Events/Extension.phpt(139) Nette\DI\Container->getService()
   in [internal function]KdybyTests\Events\ExtensionTest->testAutowire()
   in Tester/Framework/TestCase.php(143) call_user_func_array()
   in Tester/Framework/TestCase.php(50) Tester\TestCase->runTest()
   in tests/KdybyTests/bootstrap.php(36) Tester\TestCase->run()
   in KdybyTests/Events/Extension.phpt(358) run()

-- FAILED: Kdyby\Events\Extension. | KdybyTests/Events/Extension.phpt [method=testAutowireAlias]
   Exited with error code 255 (expected 0)
   E_DEPRECATED: The each() function is deprecated. This message will be suppressed on further calls
   
   in Nette/Http/RequestFactory.php(114) 
   in Tester/Framework/TestCase.php(124) call_user_func_array()
   in [internal function]Tester\TestCase->Tester\{closure}()
   in Nette/Http/RequestFactory.php(114) each()
   in cache/Nette.Configurator/Container_f70c8df11b.php(186) Nette\Http\RequestFactory->createHttpRequest()
   in Nette/DI/Container.php(179) Container_f70c8df11b->createServiceHttp__request()
   in Nette/DI/Container.php(105) Nette\DI\Container->createService()
   in cache/Nette.Configurator/Container_f70c8df11b.php(106) Nette\DI\Container->getService()
   in Nette/DI/Container.php(179) Container_f70c8df11b->createServiceApplication__application()
   in Nette/DI/Container.php(105) Nette\DI\Container->createService()
   in cache/Nette.Configurator/Container_f70c8df11b.php(98) Nette\DI\Container->getService()
   in Nette/DI/Container.php(179) Container_f70c8df11b->createServiceAlias()
   in Nette/DI/Container.php(105) Nette\DI\Container->createService()
   in KdybyTests/Events/Extension.phpt(307) Nette\DI\Container->getService()
   in [internal function]KdybyTests\Events\ExtensionTest->testAutowireAlias()
   in Tester/Framework/TestCase.php(143) call_user_func_array()
   in Tester/Framework/TestCase.php(50) Tester\TestCase->runTest()
   in tests/KdybyTests/bootstrap.php(36) Tester\TestCase->run()
   in KdybyTests/Events/Extension.phpt(358) run()


FAILURES! (60 tests, 2 failures, 2.1 seconds)

```

Yes, really early report ;)

**Comment by fprochazka** [comment]
Thank you :) Guess I should add the `nightly` to the Travis build

**Comment by remicollet** [comment]
Also noticed in Fedora QA (using 7.2.0RC6)
https://apps.fedoraproject.org/koschei/package/php-kdyby-events?collection=f28

I noticed #112 but code have change a lot from 3.1.0
So I adapt this for 3.1.0
https://src.fedoraproject.org/cgit/rpms/php-kdyby-events.git/plain/php-kdyby-events-php72.patch

Test suite passes with 5.6, 7.0, 7.1 and 7.2 (using all stable Nette components of nette 2.4)

**Comment by enumag** [comment]
@remicollet Can this be closed now? #112 is merged.
